### DISCUSSION TITLE
Use nested virtualization in CI/CD instead of a self-hosted runner

### DISCUSSION THREAD

**Discussion by dthaler** [microsoft/ebpf-for-windows#2171] [Ideas]
Today the CI/CD bottleneck seems to be contention for the limited self-hosted runner for kernel tests.

https://github.com/actions/runner-images/discussions/7191 claims nested virtualization isn't supported in github runners at present, but it is supported by Travis CI.

https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci-using-github seems to say that Travis CI runners can be used in github CI/CD.  

https://github.com/marketplace/travis-ci says Travis CI is free for open-source public repositories.

Signing up with Travis CI results in an email saying
> If you're building open-source public repositories and want to signup for an open-source plan, please [contact](mailto:support@travis-ci.com) Travis CI.

**Comment by shankarseal** [comment]
Github supports test signed runners for Windows Server 2019 and Server 2022. We will move the driver tests to github selfhost runners. This will be tracked by #2229.
